### DISCUSSION TITLE
[Question] Training Status no longer updated?

### DISCUSSION THREAD

**Discussion by gunjanmarwah** [philosowaffle/peloton-to-garmin#652] [TE / TSS / Vo2 / Intensity Minutes]
Anyone else finding today that Peloton activities synced into Garmin with HR and a solid effort are not appearing in the Training Status & Training Load? I did a 30 minute ride that I expect to be contributing to me Low Aerobic load, and a few others (even if they don't contribute; they should still appear). None are even appearing for today in the Exercise Load. Is it just me? Is it Garmin? Or are they cracking down on these imports?

**Comment by mgsimon** [comment]
I think this is resolved with #578 already. Did you try >= v4.1.0?

**Comment by gunjanmarwah** [comment]
Oh no I completely missed that. Scanned through the issues first and didn't see it. I don't think I did it right though as it's not working - are you able to help take a peek @mgsimon ? .github/workflows/sync_peloton_to_garmin.yml 
Also, any idea how far back this went? Thinking I will have to delete and resync activities.

**Comment by mgsimon** [comment]
12/21, stable image should work I think.

**Comment by gunjanmarwah** [comment]
Didn't work...
/__w/_temp/a60cc903-6e61-42cd-abe4-b3dff52e5455.sh: 1: /app/PelotonToGarminConsole: not found
Error: Process completed with exit code 127. 

I am not at all a dev, please help! :(

**Comment by gunjanmarwah** [comment]
I changed it from v3.6.1 to v4.1.0 then console-latest and then console-stable. None worked. All the same error.

**Comment by mgsimon** [comment]
I'm using docker and not sure but the run command in the master branch is ...

`    - run: /app/ConsoleClient`

Maybe there are some changes in the meantime.

**Comment by gunjanmarwah** [comment]
Got it, thanks. @philosowaffle any suggestions?

**Comment by philosowaffle** [comment]
When you setup the [Secrets](https://philosowaffle.github.io/peloton-to-garmin/latest/install/github-action/#secrets) on your Repository did you happen to setup a custom `DEVICE_INFO` secret?

**Comment by gunjanmarwah** [comment]
I did not; that is empty.

**Comment by gunjanmarwah** [comment]
@philosowaffle I've set up the Device Info now. Reran on stable, then latest, then 4.1.0. None worked, same error "/__w/_temp/9a8f2659-d352-4e8b-a69a-cdfe6ef04023.sh: 1: /app/PelotonToGarminConsole: not found
". 
Put it back to v3.6.1 and it's syncing again but still not contributing to the training load. The only thing that has really changed now is that I have Device Info probably haha.

**Comment by philosowaffle** [comment]
Alright, let's try the following.

1. Switch this [line](https://github.com/gunjanmarwah/peloton-to-garmin/blob/master/.github/workflows/sync_peloton_to_garmin.yml#L20) to use: `philosowaffle/peloton-to-garmin:console-v4`
2. Switch this [line](https://github.com/gunjanmarwah/peloton-to-garmin/blob/master/.github/workflows/sync_peloton_to_garmin.yml#L78) to use: `/app/ConsoleClient` instead

**Comment by gunjanmarwah** [comment]
@philosowaffle tried to do that. It brought in activities but they are still not showing in the Training Load.

**Comment by philosowaffle** [comment]
Ok, Let's stay on this configuration as this will atleast ensure you get future updates/fixes when they happen.

Several people noticed that after this issue was fixed, the next activity they synced earned them the `First Tacx Ride` badge on Garmin.  Did you happen see something similar? (trying to confirm your activities are getting uploaded with the correct device)

**Comment by gunjanmarwah** [comment]
Okay. No I don't see anything like that. I was hoping for these to be
synced over properly because I'm in a training plan again. Is it on the
Garmin side?

**Comment by philosowaffle** [comment]
To be honest, I'm not really sure what's going on, I just have a few more things we can try to see if anything makes a difference.  Unfortunately, this is a bit of trial and error since I'm not able to reproduce on my end and nothing stands out in the logs.

### first thing to try:
1. I think you still have a custom DeviceInfo being provided in your latest runs, let's delete that and try another sync.  I know you tried this previously, I just want to establish a baseline before trying the next step.
2. Remove DeviceInfo secret, the run a sync and see if anything changes in terms of the Device shown on Garmin and TE/TS on the workout

### next:
1. I noticed you have lines [48/49](https://github.com/gunjanmarwah/peloton-to-garmin/blob/master/.github/workflows/sync_peloton_to_garmin.yml#L48) set to True.  Let's try setting both of these to `false`.  I worry that Garmin may not be giving you credit because P2G is providing this data instead of letting Garmin calculate it.
2. Run sync again and see if anything changes.  Also let me know if you still get the "Time in Zones" charts in the workout in Garmin.

**Comment by gunjanmarwah** [comment]
Thanks I will try these shortly! For the Device Info - am I just clearing
out the secret or do I have to make any update to the code also? Previously
the secret existed but was blank. Should I revert it to that?

**Comment by philosowaffle** [comment]
Just deleting the secret from the GitHub secrets list should be sufficient. 👍

**Comment by gunjanmarwah** [comment]
Ok there's some movement! I did both things you suggested and I'm afraid I
wasn't necessarily testing the right cases as something definitely changed
but I was testing against any of ride, strengths or yoga activities. Thing
is my last ride was only Jan 8 and that was the one that made me go down
this rabbit hole.
I think this update from Garmin is looking only for activities either
tracked directly on Garmin (I had tested a strength activity tracked
directly on Garmin vs a Just Cardio on Peloton) or those with a distance. I
went all the way back to activities since and including the ride after
changing lines 48/49, deleted them, and resynced them. The ride is showing
up now in the exercise load, including the charts. None of the other
activities are showing in the load. Previously they would still show, even
if they didn't have a contribution. I'm not convinced that none of them
should have had a contribution, so I'll keep an eye on this.
I'll update you here over the coming weeks?
Thanks again for all your help!
